frontend/src/metabase/parameters/components/RequiredParamToggle/RequiredParamToggle.tsx

```diff
@@ -0,0 +1,43 @@
+import { t } from "ttag";
+import type { ReactNode } from "react";
+import { Icon, HoverCard, Stack, Flex } from "metabase/ui";
+import Toggle from "metabase/core/components/Toggle";
+import { SettingRequiredLabel } from "./RequierParamToggle.styled";
+
+interface RequiredParamToggleProps {
+  disabled?: boolean;
+  uniqueId: string;
+  value: boolean;
+  onChange: (value: boolean) => void;
+  disabledTooltip: ReactNode;
+}
+
+export function RequiredParamToggle(props: RequiredParamToggleProps) {
+  const { disabled, value, onChange, uniqueId, disabledTooltip } = props;
+  const id = `required_param_toggle_${uniqueId}`;
+
+  return (
+    <Flex gap="sm" mt="md">
+      <Toggle disabled={disabled} id={id} value={value} onChange={onChange} />
```

Comment:
Right now, the `disabled` state in new `Switch` is broken. I tried to fix it but couldn't do it quickly as it requires some sorcery with the styles. So it stays the previous component for now.